traits/has_traits.py

```diff
@@ -1037,7 +1020,7 @@ class HasTraits ( CHasTraits ):
     #-- Trait Definitions ------------------------------------------------------
 
     #: An event fired when a new trait is dynamically added to the object
-    trait_added = Event( basestring )
+    trait_added = Event( six.string_types )
```

Comment:
I think this needs to be `basestring` on Python 2 and `str` on Python 3. I'm not sure that using a `tuple` of types in an `Event` works.